server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/VacationResponseSetMethod.scala

```diff
@@ -0,0 +1,84 @@
+
+  override def process(capabilities: Set[CapabilityIdentifier],
+                       invocation: Invocation,
+                       mailboxSession: MailboxSession,
+                       processingContext: ProcessingContext): Publisher[Invocation] = {
+    asVacationResponseSetRequest(invocation.arguments)
+      .flatMap(vacationResponseSetRequest => update(mailboxSession, vacationResponseSetRequest)
+        .map(updateResult => createResponse(invocation, vacationResponseSetRequest, updateResult)))
+  }
+
+  private def update(mailboxSession: MailboxSession, vacationResponseSetRequest: VacationResponseSetRequest): SMono[UpdateResult] = {
+    vacationResponseSetRequest.parsePatch()
+        .fold(e => SMono.just(UpdateFailure(e)),
+          patch => SMono.fromPublisher(vacationRepository.modifyVacation(toVacationAccountId(mailboxSession), patch.asVacationPatch))
+            .`then`(SMono.just(UpdateSuccess())))
+  }
+
+  private def toVacationAccountId(mailboxSession: MailboxSession): org.apache.james.jmap.api.vacation.AccountId = {
+    org.apache.james.jmap.api.vacation.AccountId.fromUsername(mailboxSession.getUser)
+  }
+
+  private def asVacationResponseSetRequest(arguments: Arguments): SMono[VacationResponseSetRequest] = {
+    serializer.deserializeVacationResponseSetRequest(arguments.value) match {
+      case JsSuccess(vacationResponseSetRequest, _) => SMono.just(vacationResponseSetRequest)
+      case errors: JsError => SMono.raiseError(new IllegalArgumentException(serializer.serialize(errors).toString))
+    }
+  }
+
+  private def createResponse(invocation: Invocation,
+                             vacationResponseSetRequest: VacationResponseSetRequest,
+                             updateResult: UpdateResult): Invocation = {
+    val response = VacationResponseSetResponse(
+      accountId = vacationResponseSetRequest.accountId,
+      newState = State.INSTANCE,
+      updated = updateResult.updated,
+      notUpdated = updateResult.notUpdated)
+
+    Invocation(methodName,
+      Arguments(serializer.serialize(response).as[JsObject]),
+      invocation.methodCallId)
+  }
+
+  sealed trait UpdateResult {
+    def updated: Option[Map[String, VacationResponseUpdateResponse]]
+    def notUpdated: Option[Map[String, VacationResponseSetError]]
+  }
+  case class UpdateSuccess() extends UpdateResult {
+    override def updated: Option[Map[String, VacationResponseUpdateResponse]] = Some(Map("singleton" -> VacationResponseUpdateResponse(JsObject(Seq()))))
+
+    override def notUpdated: Option[Map[String, VacationResponseSetError]] = None
+  }
+  case class UpdateFailure(exception: Throwable) extends UpdateResult {
+    override def updated: Option[Map[String, VacationResponseUpdateResponse]] = None
+
+    override def notUpdated: Option[Map[String, VacationResponseSetError]] = Some(Map("singleton" -> asSetError))
+
+    def asSetError: VacationResponseSetError = {
+      VacationResponseSetError("", Some("error"))
+    }
+  }
```

Comment:
I prefer to put case class as top level classes, before the actual VacationResponseSetMethod